<a href="https://colab.research.google.com/github/eckoecho/CodingDojo/blob/main/Train_Test_Split_(Practice).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
from sklearn.model_selection import train_test_split
pd.set_option("display.max_columns", 100)

In [ ]:
fpath="/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week05/Data/Modified CA Housing-sklearn.csv"
df = pd.read_csv(fpath)
df.head()

MedInc  HouseAge  AveRooms  AveBedrms  Population  AveOccup  Latitude  \
0  832520.0      41.0  6.984127   1.023810       322.0  2.555556     37.88   
1  830140.0      21.0  6.238137   0.971880      2401.0  2.109842     37.86   
2  725740.0      52.0  8.288136   1.073446       496.0  2.802260     37.85   
3  564310.0      52.0  5.817352   1.073059       558.0  2.547945     37.85   
4  384620.0      52.0  6.281853   1.081081       565.0  2.181467     37.85   

   Longitude  MedHouseVal  
0    -122.23     452600.0  
1    -122.22     358500.0  
2    -122.24     352100.0  
3    -122.25     341300.0  
4    -122.25     342200.0

## Create X & y

In [ ]:
#setting our target and feature columns
X = df.drop(columns="MedHouseVal")
y = df["MedHouseVal"]

In [ ]:
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=321)

In [ ]:
#Looking at the first 5 rows of our X features
X_train[0:5]

MedInc  HouseAge  AveRooms  AveBedrms  Population  AveOccup  \
2294   246780.0      18.0  4.917157   1.086917      5466.0  2.474423   
5856   468980.0      45.0  5.039216   1.013725      1256.0  2.462745   
12427  388460.0      19.0  5.659574   1.046099      1201.0  4.258865   
4279   185760.0      37.0  2.477564   1.006410       835.0  2.676282   
19898  268750.0      23.0  6.357447   1.170213       678.0  2.885106   

       Latitude  Longitude  
2294      36.80    -119.74  
5856      34.18    -118.33  
12427     33.66    -116.21  
4279      34.09    -118.31  
19898     36.27    -119.25

In [ ]:
#Checking to ensure it is a dataframe
type(X_train)

pandas.core.frame.DataFrame

In [ ]:
#Check X_Train for null values
X_train.isna().sum()

MedInc        0
HouseAge      0
AveRooms      0
AveBedrms     0
Population    0
AveOccup      0
Latitude      0
Longitude     0
dtype: int64

In [ ]:
#Checking shape of X_train and X_test
print(f"This is X-train's shape: {X_train.shape}")
print(f"This is X-test's shape: {X_test.shape}")

This is X-train's shape: (15480, 8)
This is X-test's shape: (5160, 8)


* Which X variable has more columns?
    
    - They both have the same amount of columns.
* Which X variable has more rows?
    - The `X_train` dataset has more rows (15,480).